packages/utils/retryFn/retryFn.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * This is a function that retries an async function `numRetries` times before throwing an error.
+ *
+ * @param {Function} asyncFn A function that returns a Promise.
+ * @param {string} numRetries Default = 3.
+ */
+export default async function retryFn<T>(
+  asyncFn: () => Promise<T>,
+  numRetries = 3
```

Comment:
TypeScript can infer the type as number from the default value of 3, but i think it's better to add type nubmer

```ts
export default async function retryFn<T>(
  asyncFn: () => Promise<T>,
  numRetries: number = 3
  
  ...
```